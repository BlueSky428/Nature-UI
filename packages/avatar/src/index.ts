export * from './avatar';
export * from './avatarGroup';
