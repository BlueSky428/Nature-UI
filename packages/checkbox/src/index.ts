export * from './checkbox';
export * from './checkbox-group';
export * from './checkbox-types';
export * from './use-checkbox';
export * from './use-checkbox-group';
