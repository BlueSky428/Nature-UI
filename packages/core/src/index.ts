export * from '@nature-ui/alert';
export * from '@nature-ui/alert-dialog';
export * from '@nature-ui/avatar';
export * from '@nature-ui/breadcrumb';
export * from '@nature-ui/button';
export * from '@nature-ui/checkbox';
export * from '@nature-ui/close-button';
export * from '@nature-ui/container';
export * from '@nature-ui/counter';
export * from '@nature-ui/drawer';
export * from '@nature-ui/form-control';
export * from '@nature-ui/hooks';
export * from '@nature-ui/icon';
export * from '@nature-ui/image';
export * from '@nature-ui/input';
export * from '@nature-ui/layout';
export * from '@nature-ui/modal';
export * from '@nature-ui/popover';
export * from '@nature-ui/popper';
export * from '@nature-ui/portal';
export * from '@nature-ui/progress';
export * from '@nature-ui/radio';
export * from '@nature-ui/spinner';
export * from '@nature-ui/system';
export * from '@nature-ui/toast';
export * from '@nature-ui/transition';
export * from '@nature-ui/visually-hidden';
