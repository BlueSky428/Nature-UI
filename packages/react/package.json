{
  "name": "@nature-ui/react",
  "version": "0.1.0",
  "description": "Responsive and accessible React UI components built with react, tailwindcss and emotion",
  "keywords": [
    "react",
    "ui",
    "design-system",
    "react-components",
    "uikit",
    "accessible",
    "components",
    "emotion",
    "library",
    "design-system"
  ],
  "author": "Divine Hycenth <divinehycenth@outlook.com>",
  "homepage": "https://github.com/DNature/nature-ui",
  "license": "MIT",
  "main": "dist/index.js",
  "directories": {
    "lib": "dist",
    "test": "__tests__"
  },
  "files": [
    "dist",
    "src"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/DNature/nature-ui.git"
  },
  "scripts": {
    "build": "rimraf ./dist && tsc",
    "test": "jest --env=jsdom"
  },
  "bugs": {
    "url": "https://github.com/DNature/nature-ui/issues"
  },
  "dependencies": {
    "@nature-ui/alert": "^0.1.0",
    "@nature-ui/alert-dialog": "^0.1.0",
    "@nature-ui/avatar": "^0.1.0",
    "@nature-ui/breadcrumb": "^0.1.0",
    "@nature-ui/button": "^0.1.0",
    "@nature-ui/checkbox": "^0.1.0",
    "@nature-ui/progress": "^0.1.0",
    "@nature-ui/clickable": "^0.1.0",
    "@nature-ui/close-button": "^0.1.0",
    "@nature-ui/collapse": "^0.1.0",
    "@nature-ui/container": "^0.1.0",
    "@nature-ui/drawer": "^0.1.0",
    "@nature-ui/form-control": "^0.1.0",
    "@nature-ui/hooks": "^0.1.0",
    "@nature-ui/icon": "^0.1.0",
    "@nature-ui/image": "^0.1.0",
    "@nature-ui/input": "^0.1.0",
    "@nature-ui/layout": "^0.1.0",
    "@nature-ui/modal": "^0.1.0",
    "@nature-ui/popover": "^0.1.0",
    "@nature-ui/popper": "^0.1.0",
    "@nature-ui/portal": "^0.1.0",
    "@nature-ui/radio": "^0.1.0",
    "@nature-ui/spinner": "^0.1.0",
    "@nature-ui/system": "^0.1.0",
    "@nature-ui/toast": "^0.1.0",
    "@nature-ui/transition": "^0.1.0",
    "@nature-ui/utils": "^0.1.0",
    "@nature-ui/visually-hidden": "^0.1.0"
  },
  "devDependencies": {
    "@nature-ui/system": "^0.1.0",
    "rimraf": "^3.0.2"
  },
  "peerDependencies": {
    "@nature-ui/system": "^0.1.0",
    "react": "16.x",
    "rimraf": "^3.0.2"
  }
}
