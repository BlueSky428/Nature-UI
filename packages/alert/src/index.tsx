import * as React from 'react';
import clx from 'clsx';
import { FiInfo, FiAlertCircle, FiAlertTriangle, FiCheckCircle } from 'react-icons/fi';

const SUBTLE_TEXT = 'text-gray-800';

export const ALERT_STATUSES = {
  error: {
    bg: 'bg-red-200',
    iconColor: 'text-red-600 mr-3',
    icon: FiAlertCircle,
    variant: {
      solid: 'bg-red-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
  info: {
    bg: 'bg-blue-200',
    iconColor: 'text-blue-600 mr-3',
    icon: FiInfo,
    variant: {
      solid: 'bg-blue-600',
      subtle: SUBTLE_TEXT,
    },
  },
  success: {
    bg: 'bg-green-200',
    iconColor: 'text-green-600 mr-3',
    icon: FiCheckCircle,
    variant: {
      solid: 'bg-green-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
  warning: {
    bg: 'bg-orange-200',
    iconColor: 'text-orange-600 mr-3',
    icon: FiAlertTriangle,
    variant: {
      solid: 'bg-orange-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
};

interface AlertOptions {
  /**
   * The status of the alert
   */
  status?: keyof typeof ALERT_STATUSES;
  /**
   * The variant of the alert style to use
   */
  variant?: 'solid' | 'subtle';

  /**
   * Other tailwind utility classes or custom classnames you wish to include
   */
  className?: string;

  children?: React.ReactNode;
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component?: React.ElementType;
}

export type AlertProps = AlertOptions;

const BASE_STYLE = 'px-4 py-3 flex items-center';

const Alert: React.ForwardRefExoticComponent<AlertProps> = React.forwardRef(
  (props: AlertProps, ref: React.Ref<any>) => {
    const {
      className = '',
      status = 'success',
      children,
      variant = 'subtle',
      component: Component = 'div',
    } = props;

    // const Component = 'div';

    const VARIANT: string = ALERT_STATUSES[status].variant[variant];

    const componentClass = clx(BASE_STYLE, {
      [className]: className,
      [ALERT_STATUSES[status].bg]: status,
      [VARIANT]: variant,
    });

    const Icon = ALERT_STATUSES[status];
    const iconClasses = clx({
      [VARIANT]: variant,
      'mr-3': variant,
      [Icon.iconColor]: variant !== 'solid',
    });

    const IconComponent = Icon.icon;

    return (
      <Component className={componentClass} ref={ref}>
        {status && <IconComponent className={iconClasses} size={20} />}
        <span>{children ? children : 'This is an alert ðŸ™‚'}</span>
      </Component>
    );
  }
);

Alert.displayName = 'Alert';

export default Alert;
