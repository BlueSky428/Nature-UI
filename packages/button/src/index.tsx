import * as React from 'react';
import clx from 'clsx';

interface ButtonProps {
  /**
   * The label of the button
   */
  'aria-label'?: string;
  /**
   * The text color of the button. Use a color key passed in theme.colors.
   */
  text?: string;
  /**
   * The background color of the button. Use a color key passed in theme.colors.
   */
  color?: string;
  /**
   * The variant of the button style to use.
   */
  variant?: 'outline' | 'ghost' | 'link' | 'solid' | 'none';
  /**
   * If true, the button will be disabled.
   */
  isDisabled?: boolean;
  /**
   * If true, the button will show a spinner.
   */
  isLoading?: boolean;
  /**
   * The label to show in the button when isLoading is true. If no text is passed, it only shows the spinner
   */
  loadingText?: string;
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'none';
  /**
   * Other tailwind utility classes or custom classnames you wish to include
   */
  className?: string;
  /**
   * The lightness or how dark you want the background color to be
   */
  children?: React.ReactNode;
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   * defaults to "button"
   */
  component?: React.ElementType;
  id?: string;
  /**
   *  when the href prop is set it changes the tag name to a
   */
  href?: string;
  onClick?: () => any;
}

const darken = (value: string): string => {
  const splitStr = value.split('-');
  const dept = Number(splitStr[1]);
  const color = splitStr[0];

  if (dept >= 800) {
    return `${color}-${dept - 200}`;
  }

  return `${color}-${dept + 200}`;
};

const lighten = (value: string): string => {
  const color = value.split('-')[0];

  return `${color}-${100}`;
};

const Button = React.forwardRef((props: ButtonProps, ref: React.Ref<any>) => {
  const {
    component: Component = 'button',
    variant = 'solid',
    color = 'teal-500',
    text = 'white',
    size = 'md',
    children,
    ...rest
  } = props;
  const DEFAULT_CLASS = 'block focus:shadow-outline focus:outline-none rounded font-semibold flex';
  const STYLES = {
    solid: `bg-${color} text-${text} hover:bg-${darken(color)}`,
    outline: `bg-transparent hover:bg-${lighten(
      text
    )} text-${text} border border-${text} focus:border-transparent`,
    ghost: `hover:bg-${lighten(text)} text-${text}`,
    link: `hover:underline text-${text}`,
    md: 'px-4 py-2',
  };

  let BTNClass: string;

  if (variant === 'none') {
    BTNClass = '';
  } else {
    BTNClass = clx(DEFAULT_CLASS, {
      [STYLES[size]]: size && variant !== 'link',
      [STYLES[variant]]: variant,
    });
  }

  return (
    <Component className={BTNClass} ref={ref} {...rest}>
      {children ? children : 'Button'}
    </Component>
  );
});

export default Button;
