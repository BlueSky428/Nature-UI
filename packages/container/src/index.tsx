import * as React from 'react';
import clx from 'clsx';

interface IContainer {
  children?: React.ReactNode;
  /**
   * Determine the max-width of the container.
   */
  size?: 'lg' | 'md' | 'sm' | 'xl' | 'xs';
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component?: React.ComponentType;
  className?: string;
}

const BASE_STYLE = 'mx-auto container';
const LG = 'px-20';
const MD = 'px-32';
const SM = 'px-48';
const XL = 'px-0';
const XS = 'px-64';

const Container: React.ForwardRefExoticComponent<IContainer> = React.forwardRef(
  (props: IContainer, ref: React.Ref<any>) => {
    const { component: Component = 'div', size = 'lg', children, className = '' } = props;

    const componentClass: string = clx(BASE_STYLE, {
      [LG]: size === 'lg',
      [MD]: size === 'md',
      [SM]: size === 'sm',
      [XL]: size === 'xl',
      [XS]: size === 'xs'
    });

    return (
      <Component className={componentClass} ref={ref}>
        <Component className={className}>{children}</Component>
      </Component>
    );
  }
);

Container.displayName = 'Container';

export default Container;
