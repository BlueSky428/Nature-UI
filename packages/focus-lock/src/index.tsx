import * as React from 'react';
import { getAllFocusable, isFocusable, __DEV__ } from '@nature-ui/utils';
import ReactFocusLock from 'react-focus-lock';

export interface FocusLockProps {
  /**
   * `ref` of the element to receive focus initially
   */
  initialFocusRef?: React.RefObject<HTMLElement>;
  /**
   * `ref` of the element to return focus to when `FocusLock`
   * unmounts
   */
  finalFocusRef?: React.RefObject<HTMLElement>;
  /**
   * The `ref` of the wrapper for which the focus-lock wraps
   */
  contentRef?: React.RefObject<HTMLElement>;
  /**
   * If `true`, focus will be restored to the element that
   * triggered the `FocusLock` once it unmounts
   */
  restoreFocus?: boolean;
  /**
   * The component to render
   */
  children: React.ReactNode;
  /**
   * If `true`, focus trapping will be disabled
   */
  isDisabled?: boolean;
  /**
   * If `true`, the first focuable element within the `children`
   * will ne auto-focused once `FocusLock` mounts
   */
  autoFocus?: boolean;
}

/**
 * React component to trap focus within an element or component.
 * Mostly used in Modals, Popovers, etc.
 */
export const FocusLock = (props: FocusLockProps) => {
  const {
    initialFocusRef,
    finalFocusRef,
    contentRef,
    restoreFocus,
    children,
    isDisabled,
    autoFocus = true,
  } = props;

  const onActivation = React.useCallback(() => {
    if (initialFocusRef?.current) {
      initialFocusRef.current.focus();
    } else {
      if (contentRef?.current) {
        const focusable = getAllFocusable(contentRef.current);
        if (focusable.length === 0) {
          contentRef?.current?.focus();
        }
      }
    }
  }, [initialFocusRef, contentRef]);

  const onDeactivation = React.useCallback(() => {
    finalFocusRef?.current?.focus();
  }, [finalFocusRef]);

  const returnFocus = restoreFocus && !finalFocusRef;

  return (
    <ReactFocusLock
      autoFocus={autoFocus}
      disabled={isDisabled}
      onActivation={onActivation}
      onDeactivation={onDeactivation}
      returnFocus={returnFocus}
    >
      {children}
    </ReactFocusLock>
  );
};

if (__DEV__) {
  FocusLock.displayName = 'FocusLock';
}

export default FocusLock;
