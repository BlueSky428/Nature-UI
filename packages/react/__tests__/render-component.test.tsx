import * as React from 'react';
import { render } from '@nature-ui/test-utils';

import { Alert } from '../src';

describe('@nature-ui/react', () => {
  test('it should render Alert component', () => {
    const tools = render(
      <Alert alertTitle='Alert title'>Alert description</Alert>,
    );

    expect(tools.asFragment()).toMatchSnapshot();
  });
});
