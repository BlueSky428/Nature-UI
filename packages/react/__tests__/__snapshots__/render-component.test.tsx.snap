// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nature-ui/react it should render Alert component 1`] = `
<DocumentFragment>
  <div
    class="px-4 py-3 flex items-center w-full relative overflow-hidden bg-red-200 text-gray-800"
    role="alert"
  >
    <svg
      class="text-gray-800 mr-3 text-red-600 mr-3"
      fill="none"
      height="20"
      size="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <line
        x1="12"
        x2="12"
        y1="8"
        y2="12"
      />
      <line
        x1="12"
        x2="12"
        y1="16"
        y2="16"
      />
    </svg>
    <div
      class="mr-3 font-bold"
    >
      Alert title
    </div>
    <div
      class=""
    >
      Alert description
    </div>
  </div>
</DocumentFragment>
`;
